statsmodels/tsa/arima/estimators/gls.py

```diff
@@ -0,0 +1,305 @@
+"""
+Feasible generalized least squares for regression with SARIMA errors.
+
+Author: Chad Fulton
+License: BSD-3
+"""
+import numpy as np
+import warnings
+
+from statsmodels.tools.tools import add_constant, Bunch
+from statsmodels.regression.linear_model import OLS
+from statsmodels.tsa.innovations import arma_innovations
+from statsmodels.tsa.statespace.tools import diff
+
+from statsmodels.tsa.arima.estimators.yule_walker import yule_walker
+from statsmodels.tsa.arima.estimators.burg import burg
+from statsmodels.tsa.arima.estimators.hannan_rissanen import hannan_rissanen
+from statsmodels.tsa.arima.estimators.innovations import (
+    innovations, innovations_mle)
+from statsmodels.tsa.arima.estimators.statespace import statespace
+
+from statsmodels.tsa.arima.specification import SARIMAXSpecification
+from statsmodels.tsa.arima.params import SARIMAXParams
+
+
+def gls(endog, exog=None, order=(0, 0, 0), seasonal_order=(0, 0, 0, 0),
+        include_constant=None, n_iter=None, max_iter=50, tolerance=1e-8,
+        arma_estimator='innovations_mle', arma_estimator_kwargs=None):
+    """
+    Estimate ARMAX parameters by GLS.
+
+    Parameters
+    ----------
+    endog : array_like
+        Input time series array.
+    exog : array_like, optional
+        Array of exogenous regressors. If not included, then `include_constant`
+        must be True, and then `exog` will only include the constant column.
+    order : tuple, optional
+        The (p,d,q) order of the ARIMA model. Default is (0, 0, 0).
+    seasonal_order : tuple, optional
+        The (P,D,Q,s) order of the seasonal ARIMA model.
+        Default is (0, 0, 0, 0).
+    include_constant : bool, optional
+        Whether to add a constant term in `exog` if it's not already there.
+        The estimate of the constant will then appear as one of the `exog`
+        parameters. If `exog` is None, then the constant will represent the
+        mean of the process. Default is True if the specified model does not
+        include integration and False otherwise.
+    n_iter : int, optional
+        Optionally iterate feasible GSL a specific number of times. Default is
+        to iterate to convergence. If set, this argument overrides the
+        `max_iter` and `tolerance` arguments.
+    max_iter : int, optional
+        Maximum number of feasible GLS iterations. Default is 50. If `n_iter`
+        is set, it overrides this argument.
+    tolerance : float, optional
+        Tolerance for determining convergence of feasible GSL iterations. If
+        `iter` is set, this argument has no effect.
+        Default is 1e-8.
+    arma_estimator : string, optional
+        The estimator used for estimating the ARMA model. This option should
```

Comment:
This is a pretty big block of text.  Might want to format with sub-sections for the lists?

Reply:
Good idea - lets see if I can get the RST formatting right...